generator/php/build.gradle

```diff
@@ -48,39 +49,45 @@ task generateClient(type: GradleBuild) {
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
+        
+        def artifactVersion = versionCategory == 'preview' ? '0' : criteoApiVersion.replaceAll("_",".").toString()
+        artifactVersion += ".${generatorVersion}.${getFormattedDate()}".toString() // example : 2021.01.0.211110 for stable , 0.0.211110 for preview
 
-        def generateTask = task("openApiGenerate_php_" + parsedApiName, type: GenerateTask.class) {
+        def generateTask = task("openApiGenerate_${technologyStack}_" + parsedApiName, type: GenerateTask.class) {
                 configFile = "$projectDir/generatorConfiguration.yaml".toString()
-                generatorName = "php"
+                generatorName = "$language".toString()
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        invokerPackage          : "${packageBase}".toString(),
-                        packageUrl              : 'https://github.com/criteo/criteo-api-php-sdk',
+                        groupId                 : 'criteo',
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
+                        artifactVersion         : "${artifactVersion}".toString(),
+                        description             : "${language.toUpperCase()} SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
                         hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211110
-                        groupId                 :  'criteo',
+                        invokerPackage          : "${namespaceBase}".toString(),
+                        packageName             : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
+                        packageUrl              : "https://github.com/criteo/criteo-api-${language}-sdk".toString(),
```

Comment:
For PHP SDKs there will be 2 repositories, one for each CriteoService:
- [criteo-api-retailmedia-php-sdk](https://github.com/criteo/criteo-api-retailmedia-php-sdk)
- [criteo-api-marketingsolutions-php-sdk](https://github.com/criteo/criteo-api-marketingsolutions-php-sdk)
So I think the string template should more be:
`"https://github.com/criteo/criteo-api-${criteoService}-${language}-sdk".toString()`

Reply:
We will update the story and take these latest decision into account in next commit